src/basic/forward.h

```diff
@@ -0,0 +1,303 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include <errno.h>              // IWYU pragma: export
+#include <inttypes.h>           // IWYU pragma: export
+#include <limits.h>             // IWYU pragma: export
+#include <stdarg.h>             // IWYU pragma: export
+#include <stdbool.h>            // IWYU pragma: export
+#include <stddef.h>             // IWYU pragma: export
+#include <stdint.h>             // IWYU pragma: export
+#include <sys/types.h>          // IWYU pragma: export
+#include <uchar.h>              // IWYU pragma: export
+
+#include "assert-util.h"        // IWYU pragma: export
+#include "cleanup-util.h"       // IWYU pragma: export
+#include "macro.h"              // IWYU pragma: export
```

Comment:
Why this includes so many headers?? Are they all really necessary?

Reply:
These all provide stuff that is often used in headers and gets pulled in one way or another regardless for each source file. So instead of repeating these includes all over the place, we centralize them in one file so we only have to include forward.h and nothing else.